node_modules/col-whiteboards/lib/api.js

```diff
@@ -400,26 +416,47 @@ var editWhiteboard = module.exports.editWhiteboard = function(ctx, id, title, me
         return callback(err);
       }
 
-      // Update the whiteboard in the DB
-      var update = {
-        'title': title
-      };
-
-      whiteboard.updateAttributes(update).complete(function(err, whiteboard) {
+      // Get the whiteboard members
+      whiteboard.getUsers({'attributes': UserConstants.BASIC_USER_FIELDS}).complete(function(err, oldMembers) {
         if (err) {
-          log.error({'err': err, 'id': id}, 'Failed to update a whiteboard');
+          log.error({'err': err, 'whiteboard': id}, 'Failed to get the whiteboard members');
           return callback({'code': 500, 'msg': err.message});
         }
 
-        // Update the members of the whiteboard
-        whiteboard.setUsers(members).complete(function(err) {
+        // Update the whiteboard in the DB
+        var update = {
+          'title': title
+        };
+        whiteboard.updateAttributes(update).complete(function(err, whiteboard) {
           if (err) {
-            log.error({'err': err}, 'Failed to update the members of a whiteboard');
+            log.error({'err': err, 'id': id}, 'Failed to update a whiteboard');
             return callback({'code': 500, 'msg': err.message});
           }
 
-          // Retrieve the updated whiteboard, including the members and elements
-          return getWhiteboardProfile(ctx, whiteboard.id, callback);
+          // Update the members of the whiteboard
+          whiteboard.setUsers(members).complete(function(err) {
+            if (err) {
+              log.error({'err': err}, 'Failed to update the members of a whiteboard');
+              return callback({'code': 500, 'msg': err.message});
+            }
+
+            // Retrieve the updated whiteboard, including the members and elements
+            getWhiteboardProfile(ctx, whiteboard.id, function(err, whiteboardProfile) {
+              if (err) {
+                return callback(err);
+              }
+
+              callback(null, whiteboardProfile);
+
+              // Get the new members
+              oldMembers = _.indexBy(oldMembers, 'id');
```

Comment:
We still need the full user objects, I think this is slightly easier to ready (and minimally better performant)